sympy/solvers/solveset.py

```diff
@@ -3395,7 +3395,7 @@ def nonlinsolve(system, *symbols):
 
         # positive dimensional system
         res = _handle_positive_dimensional(polys, symbols, denominators)
-        if isinstance(res, EmptySet) and any(not p.domain.is_Exact for p in polys):
+        if isinstance(res, type(EmptySet)) and any(not p.domain.is_Exact for p in polys):
```

Comment:
Same here

Reply:
Yes. I've checked that there are no other cases using `type(EmptySet)`.